src/actions/index.js

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+
+const loading = username=> ({ type: 'LOADING', payload: username });
+
+const loadResult = (repos) => ({ 
+    type: 'LOAD_RESULT',
+    payload: repos
+});
+
+export const getResult = searchTerm => {
+    return async dispatch => {
+        dispatch(loading(searchTerm));
+        try {
+            const repos = await fetchRepos(searchTerm);
+            console.log(repos)
+            dispatch(loadResult(repos))
+        } catch (err) {
+            console.warn(err.message);
+            dispatch({ type: 'SET_ERROR', payload: err.message })
+        };
+    };
+};
+export const fetchRepos = async(username) =>{
+    try {
+        const resp = await axios.get(`https://api.github.com/users/${username}/repos`)
+        const data = resp.data
+        console.log(data)
+        return data
+    }
+    catch(err) {
```

Comment:
To change the error message when the username isn't found you can use:
catch (error) {
        if(error.response.status === 404){throw new Error('Not a valid username!')}
        throw Error(error.message);
    }